### PR TITLE
Parsers: Fluentd Syslog Support #437

### PR DESCRIPTION
**Pull request by sardev08**
implement fluentd syslog parsers

## Background
https://github.com/panther-labs/panther/issues/437

## Changes

implemented fluentd syslog parsers for RFC3164 and 5424

## Testing
Unit tests

### CODE REVIEW

File: internal/log_analysis/log_processor/parsers/fluentdsysloglogs/rfc3164.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
+package fluentdsysloglogs
```
Comment (jacknagz) [inline internal/log_analysis/log_processor/parsers/fluentdsysloglogs/rfc3164.go]: Can we just call this package `fluentdsyslogs` cc @kostaspap / @rleighton

File: internal/log_analysis/log_processor/parsers/fluentdsysloglogs/rfc3164.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+	"github.com/panther-labs/panther/internal/log_analysis/log_processor/parsers"
+	"go.uber.org/zap"
+)
+
+var RFC3164Desc = `Fluentd syslog parser for the RFC3164 format (ie. BSD-syslog messages)
+Reference: https://docs.fluentd.org/parser/syslog#rfc3164-log`
+
+// nolint:lll
+type RFC3164 struct {
+	Priority *uint8  `json:"pri" validate:"required" description:"Priority is calculated by (Facility * 8 + Severity). The lower this value, the higher importance of the log message."`
```
Comment (rleighton) [inline internal/log_analysis/log_processor/parsers/fluentdsysloglogs/rfc3164.go]: Does this log NOT have some sort for a timestamp? I find that odd. According to the link above there is a timestamp. We need that.

File: internal/log_analysis/log_processor/parsers/fluentdsysloglogs/rfc5424.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+	jsoniter "github.com/json-iterator/go"
+	"github.com/panther-labs/panther/internal/log_analysis/log_processor/parsers"
+	"go.uber.org/zap"
+)
+
+var RFC5424Desc = `Fluentd syslog parser for the RFC3164 format (ie. BSD-syslog messages)
+Reference: https://docs.fluentd.org/parser/syslog#rfc3164-log`
+
+// nolint:lll
+type RFC5424 struct {
```
Comment (rleighton) [inline internal/log_analysis/log_processor/parsers/fluentdsysloglogs/rfc5424.go]: Same comment as above, we need the timetsamp.

File: internal/log_analysis/log_processor/parsers/fluentdsyslogs/rfc3164_test.go [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+	checkRFC3164(t, log, expectedRFC3164)
+}
+
+func TestRFC3164TypeType(t *testing.T) {
+	parser := &RFC3164Parser{}
+	require.Equal(t, "Fluentd.Syslog3164", parser.LogType())
+}
+
+func checkRFC3164(t *testing.T, log string, expectedRFC3164 *RFC3164) {
+	parser := &RFC3164Parser{}
```
Comment (rleighton) [inline internal/log_analysis/log_processor/parsers/fluentdsyslogs/rfc3164_test.go]: This was not merged with master. Please pull master and the missing SetEvent()

File: internal/log_analysis/log_processor/parsers/timestamp/timestamp.go [hunk 5]
```diff
@@ -104,3 +106,22 @@ func (ts *UnixMillisecond) UnmarshalJSON(jsonBytes []byte) (err error) {
 	*ts = (UnixMillisecond)(t)
 	return nil
 }
+
+type FluentdTimestamp time.Time
```
Comment (rleighton) [inline internal/log_analysis/log_processor/parsers/timestamp/timestamp.go]: Please add unit tests.

**Pull request merged**